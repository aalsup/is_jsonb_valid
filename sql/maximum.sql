-- maximum validation
-- below the maximum is valid
SELECT is_jsonb_valid('{"maximum":3}', '2.6');
-- boundary point is valid
SELECT is_jsonb_valid('{"maximum":3}', '3');
-- above the maximum is invalid
SELECT is_jsonb_valid('{"maximum":3}', '3.5');
-- ignores non-numbers
SELECT is_jsonb_valid('{"maximum":3}', '"x"');
-- exclusiveMaximum validation
-- below the maximum is still valid
SELECT is_jsonb_valid('{"maximum":3,"exclusiveMaximum":true}', '2.2');
-- boundary point is invalid
SELECT is_jsonb_valid('{"maximum":3,"exclusiveMaximum":true}', '3');