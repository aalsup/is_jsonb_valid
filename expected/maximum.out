-- maximum validation
-- below the maximum is valid
SELECT is_jsonb_valid('{"maximum":3}', '2.6');
 is_jsonb_valid 
----------------
 t
(1 row)

-- boundary point is valid
SELECT is_jsonb_valid('{"maximum":3}', '3');
 is_jsonb_valid 
----------------
 t
(1 row)

-- above the maximum is invalid
SELECT is_jsonb_valid('{"maximum":3}', '3.5');
 is_jsonb_valid 
----------------
 f
(1 row)

-- ignores non-numbers
SELECT is_jsonb_valid('{"maximum":3}', '"x"');
 is_jsonb_valid 
----------------
 t
(1 row)

-- exclusiveMaximum validation
-- below the maximum is still valid
SELECT is_jsonb_valid('{"maximum":3,"exclusiveMaximum":true}', '2.2');
 is_jsonb_valid 
----------------
 t
(1 row)

-- boundary point is invalid
SELECT is_jsonb_valid('{"maximum":3,"exclusiveMaximum":true}', '3');
 is_jsonb_valid 
----------------
 f
(1 row)

